src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/simple/RandomizedJoinPlanOptimizerImpl.java

```diff
@@ -0,0 +1,91 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.simple;
+
+import java.util.List;
+import java.util.Random;
+
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.utils.PhysicalPlanFactory;
+import se.liu.ida.hefquin.engine.queryproc.QueryOptimizationException;
+
+public class RandomizedJoinPlanOptimizerImpl extends JoinPlanOptimizerBase {
+	@Override // Implement initializeEnumerationAlgorithm from JoinPlanOptimizerBase.
+	public EnumerationAlgorithm initializeEnumerationAlgorithm( final List<PhysicalPlan> subplans ) {
+		return new RandomizedEnumerationAlgorithm(subplans);
+	}
+	
+	protected class RandomizedEnumerationAlgorithm implements EnumerationAlgorithm {
+		
+		/**
+		 * The class contains a random seed object.
+		 */
+		Random random;
+		
+		/**
+		 * Have a list of subplans.
+		 */
+		protected final List<PhysicalPlan> subplans;
+
+		/**
+		 *  When initializing the algorithm, start with the given list of subplans.
+		 *  @param subplans
+		 */
+		public RandomizedEnumerationAlgorithm( final List<PhysicalPlan> subplans ) {
+			this.subplans = subplans;
+			this.random = new Random(); // Generates the random seed.
+		}
+		
+
+		/**
+		 * Also taken from Greedy and left unchanged save for the different method being called within.
+		 */
+		@Override
+		public PhysicalPlan getResultingPlan() throws QueryOptimizationException {
+			PhysicalPlan currentPlan = chooseFirstSubplan();
+
+			while ( subplans.size() > 0 ){
+				currentPlan = addNextRandomJoin(currentPlan);
+			}
+
+			return currentPlan;
+		}
+		
+
+		/**
+		 * Chooses the first subplan.
+		 * Instead of the best plan chosen by Greedy, this method picks a completely random plan.
+		 * @return
+		 * @throws QueryOptimizationException
+		 */
+		protected PhysicalPlan chooseFirstSubplan() throws QueryOptimizationException {
+			int indexOfRandomPlan = this.random.nextInt(subplans.size()); // nextInt returns an int between 0 (inclusive) and size (exclusive).

```

Comment:
Two things here:
1. When you are referring to member variables, you don't need to prefix the member variables with `this.`. While doing so is possible, it is not a common practice in Java.
2. What I mentioned above about `final` also holds for local variables such as this one. Unless a local variable is explicitly meant to be reassigned within the method, it should be marked `final` as a safety measure.

Hence, the line should be changed as follows:
```suggestion
			final int indexOfRandomPlan = random.nextInt(subplans.size()); // nextInt returns an int between 0 (inclusive) and size (exclusive).
```